### PR TITLE
feat(suite): add transports menu in debug

### PR DESCRIPTION
**Pull request by mroz22**
part of part of https://github.com/trezor/trezor-suite/pull/6509. needs to be based on https://github.com/trezor/trezor-suite/pull/7411

this is not finished but since I would like to discuss few todos I have left in the code I am moving this into 'to be reviewed' column

motivation: 
- at the moment, bridge is spawned automatically whenever suite starts. with the advent of nodeusb, which I believe we wan't to keep somewhere behind debug flag for testing, we need a way how to override this behaviour.

adds: 
- way to stop/start bridge from debug UI
- customize TrezorConnect.init transports parameter from debug UI 

possible improvements/followups:
- create a generalized way how to start/stop other process in suite
- creeate a way how to restart trezor connect without restarting suite

image

### CODE REVIEW

File: packages/suite/src/views/settings/debug/Transport.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    useEffect(() => {
+        TrezorConnect.getSettings().then(result => {
+            if (!result.success) return;
+            setConnectSettings(result.payload);
+        });
+    }, []);
+
+    const { debug } = useSelector(state => ({
+        debug: state.suite.settings.debug,
+        activeTransport: state.suite.transport,
```
Comment (marekrjpolak) [inline packages/suite/src/views/settings/debug/Transport.tsx]: Unused?

File: packages/suite/src/views/settings/debug/Transport.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }));
+
+    // fallback [] to avoid need of migration.
+    const debugTransports = debug.transports || [];
+
+    const transports: TransportMenuItem[] = useMemo(() => {
+        const transports: TransportMenuItem[] = [
+            {
+                name: 'BridgeTransport',
```
Comment (marekrjpolak) [inline packages/suite/src/views/settings/debug/Transport.tsx]: nit: they probably don't have to be objects given that you're making them objects at the end of useMemo.

File: packages/suite/src/views/settings/debug/Transport.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                            onClick={() => {
+                                let nextTransports = [...debugTransports];
+                                if (nextTransports.includes(transport.name)) {
+                                    nextTransports.splice(
+                                        debugTransports.findIndex(t => t === transport.name),
+                                        1,
+                                    );
+                                } else {
+                                    nextTransports = [...debugTransports, transport.name];
+                                }
```
Comment (marekrjpolak) [inline packages/suite/src/views/settings/debug/Transport.tsx]: nit: how 'bout
```suggestion
                                const nextTransports = debugTransports.includes(transport.name)
                                    ? debugTransports.filter(t => t !== transport.name)
                                    : [...debugTransports, transport.name];
```

File: packages/suite/src/views/settings/debug/Processes.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+        const onBridgeProcessEvent = (status: Process) => {
+            setBridgeProcess(status);
+        };
+        desktopApi.on('bridge/status', onBridgeProcessEvent);
+
+        // todo: how to remove event listener from desktopApi?
+        // return () => {
+        //     desktopApi.remove('bridge/status', onBridgeProcessEvent);
+        // };
```
Comment (marekrjpolak) [inline packages/suite/src/views/settings/debug/Processes.tsx]: Currently only this, unfortunately, but should be ok here.
```suggestion
        return () => {
            desktopApi.removeAllListeners('bridge/status');
        };
```

File: packages/suite-desktop-api/src/api.ts [hunk 5]
```diff
@@ -69,6 +73,8 @@ export interface InvokeChannels {
     'metadata/write': (options: { file: string; content: string }) => InvokeResult;
     'server/request-address': (route: string) => string | undefined;
     'tor/toggle': (shouldEnableTor: boolean) => InvokeResult;
+    'bridge/toggle': () => InvokeResult;
+    'bridge/get-status': () => InvokeResult;
```
Comment (marekrjpolak) [inline packages/suite-desktop-api/src/api.ts]: If there is a payload, it should be `() => InvokeResult<Status>` instead, no?

File: packages/suite-desktop/src/modules/bridge.ts [hunk 6]
```diff
@@ -20,6 +34,37 @@ const load = async () => {
...
+            return { success: false, error };
+        } finally {
+            handleBridgeStatus(bridge, mainWindow);
+        }
+    });
+
+    ipcMain.handle('bridge/get-status', async () => {
+        try {
+            const status = await bridge.status();
+            return { success: true, status };
```
Comment (marekrjpolak) [inline packages/suite-desktop/src/modules/bridge.ts]: If type would be corrected, this'll be `{ success: true, payload: status }` instead.

Also, in equivalent code in tor module we don't return a value from `tor/get-status` but emit `tor/status` instead, which allows us to handle the status in exactly the same manner whether it was demanded or not. I'm not saying it's necessarily better, but let's discuss it.

File: packages/suite-desktop/src/modules/bridge.ts [hunk 7]
```diff
@@ -20,6 +34,37 @@ const load = async () => {
         bridge.stop();
     });
 
+    ipcMain.handle('bridge/toggle', async (_: unknown) => {
+        const status = await handleBridgeStatus(bridge, mainWindow);
```
Comment (marekrjpolak) [inline packages/suite-desktop/src/modules/bridge.ts:36]: In a single `bridge/toggle`, `bridge/status` will be emitted twice. Is it expected and/or required?

File: packages/suite/src/views/settings/debug/Transport.tsx (hunk 1 above)
Comment (mroz22) [reply packages/suite/src/views/settings/debug/Transport.tsx]: removed in af9f722aed67c3a268995e340867bd9d37752c3d

File: packages/suite/src/views/settings/debug/Transport.tsx (hunk 2 above)
Comment (mroz22) [reply packages/suite/src/views/settings/debug/Transport.tsx]: sure, why not af9f722aed67c3a268995e340867bd9d37752c3d

File: packages/suite/src/views/settings/debug/Transport.tsx (hunk 3 above)
Comment (mroz22) [reply packages/suite/src/views/settings/debug/Transport.tsx]: cool af9f722aed67c3a268995e340867bd9d37752c3d

File: packages/suite/src/views/settings/debug/Processes.tsx (hunk 4 above)
Comment (mroz22) [reply packages/suite/src/views/settings/debug/Processes.tsx]: top af9f722aed67c3a268995e340867bd9d37752c3d

File: packages/suite-desktop-api/src/api.ts (hunk 5 above)
Comment (mroz22) [reply packages/suite-desktop-api/src/api.ts]: thanks, fixed in 2b92f68103839b87ef3e5996f8d1a686b2235eda

File: packages/suite-desktop/src/modules/bridge.ts (hunk 6 above)
Comment (mroz22) [reply packages/suite-desktop/src/modules/bridge.ts]: fixed in 2b92f68103839b87ef3e5996f8d1a686b2235eda

I somewhat dislike that `something/get-something` triggers firing of event instead of returnig something directly :thinking: hard to follow. But I am open to discuss it in the office

File: packages/suite-desktop/src/modules/bridge.ts (hunk 7 above)
Comment (mroz22) [reply packages/suite-desktop/src/modules/bridge.ts:36]: not required but it does not hurt anybody. also not nice. again, open to discuss it in the office.

**Review by marekrjpolak — APPROVED** [review]
Approved!
